asapdiscovery-docking/asapdiscovery/docking/docking_v2.py

```diff
@@ -0,0 +1,283 @@
+import abc
+from enum import Enum
+from pathlib import Path
+from typing import Literal, Optional, Union
+
+import dask
+from asapdiscovery.data.dask_utils import actualise_dask_delayed_iterable
+from asapdiscovery.data.openeye import (
+    combine_protein_ligand,
+    oechem,
+    oedocking,
+    oeomega,
+    save_openeye_pdb,
+)
+from asapdiscovery.data.schema_v2.ligand import Ligand, compound_names_unique
+from asapdiscovery.data.schema_v2.pairs import DockingInputPair
+from asapdiscovery.docking.docking_data_validation import DockingResultCols
+from asapdiscovery.modeling.modeling import split_openeye_design_unit
+from pydantic import BaseModel, Field, PositiveFloat, PositiveInt, root_validator
+
+
+class SCORE_TYPE(Enum):
+    CHEMGAUSS4 = "chemgauss4"
+
+
+class DockingResult(BaseModel):
+    """
+    Schema for a DockingResult, containing both a DockingInputPair used as input to the workflow
+    and a Ligand object containing the docked pose.
+    Also contains the probability and chemgauss4 score of the docked pose.
+    """
+
+    input_pair: DockingInputPair = Field(description="Input pair")
```

Comment:
Just want to check I understand is the input pair the receptor and a Ligand object and then the posed ligand is the ligand again but with a different geometry?

Reply:
The input `pair` is a combination of:

- A query `Ligand` to dock to the receptor.
- A `PreppedComplex` that has the receptor + bound ligand (if present) as an `oedu` AND a `Ligand` that represents the compound originally bound to the receptor (if present). This allows us to do matching against the receptor for chemoinf easily. 

Yes the `posed_ligand` is then the same query ligand again in a different conformation.